### DISCUSSION TITLE
[Feature] Provide onclick event (function) for match

### DISCUSSION THREAD

**Discussion by yongshun950824** [Drarig29/brackets-viewer.js#80] [Ideas]
### Feature: 
Provide on-click event (function) for match.

### Description:
When clicking the match / or any team (within the redbox as the attached screenshot), able to perform the action as provided Javascript/Typescript function.

![Sample Tournament Bracket with highlighted (red box) match](https://user-images.githubusercontent.com/29250581/154626171-ebd472c0-9ab7-4c55-a45a-d48a98632f92.png)

### Coding & Concept:

Settings for bracketsViewer

```typescript
window.bracketsViewer.render({
    stages: data.stage,
    matches: data.match,
    matchGames: data.match_game,
    participants: data.participant,
}, {
    selector: '#example',
    ...  // Other settings
    onMatchClick: onMatchClick // Add new onclick event with pass the method setting
})
```

Method to be passed with expected to receive `match` as an input parameter.

```typescript
onMatchClick(match) {
   // Following implementation
}
```

Able to get the `match` info as `Match` (class) structure.

```JSON
{
    "id": 0,
    "number": 1,
    "stage_id": 0,
     "group_id": 0,
     "round_id": 0,
     "child_count": 0,
     "status": 4,
     "opponent1": {
         "id": 0,
         "position": 1,
         "score": 16,
         "result": "win"
     },
     "opponent2": {
        "id": 1,
        "position": 2,
        "score": 12,
        "result": "loss"
    }
}
```

### Ideas: 

1. Toornament Bracket & Match flow [Reference: [http://www.toornament.com])

Flow:
a. Toornament Bracket Match to be clicked.
![Toornament Bracket Match to be clicked](https://user-images.githubusercontent.com/29250581/154627551-54a68651-c31a-4ef3-8f92-eb81739005c6.png)

b. After the click, redirect to next page.
![After the click, redirect to next page (Toornament Match Details)](https://user-images.githubusercontent.com/29250581/154627595-12a313cf-3c26-4b80-b6fc-297cc4ef5ad6.png)

2. jQuery Bracket on-mouse click & on-mouse hover [Reference: [http://www.aropupu.fi/bracket/#matches])

**Comment by Drarig29** [comment]
Actually the onclick event already exists:

- Implemented [here](https://github.com/Drarig29/brackets-viewer.js/blob/ee61103c80dfea8c294aa69a05e6ee578482a71d/src/main.ts#L33)
- Example of use [here](https://github.com/Drarig29/brackets-viewer.js/blob/ee61103c80dfea8c294aa69a05e6ee578482a71d/demo/with-api.html#L41)

Currently it's a class property so you need to set it directly. But I like your idea of passing it as a prop in the settings.

I won't do it for now because I don't have the time and it's not a high priority issue since it already exists.

**Reply by yongshun950824** [reply]
Okay nice one, the feature is enough for me.

Thanks for the assists. =)

**Reply by Drarig29** [reply]
Fixed in https://github.com/Drarig29/brackets-viewer.js/issues/81

**Comment by KaranCK0** [comment]
![image (1)](https://github.com/user-attachments/assets/ae43d10d-5fb4-4740-af21-380adc6bc807)
How we put this label in brackets.

**Reply by Drarig29** [reply]
Hi! What do you want to do? Change them?

**Reply by KaranCK0** [reply]
I want a label inside the next matches.
You can see inside the image I already marked it.

**Reply by Drarig29** [reply]
I made the choice to not have labels when the source of the participant is obvious thanks to the gray line connectors, directly connected to the previous match. Those labels are only present in double elimination brackets, for connections between the winner bracket and the loser bracket which are not obvious at all.

**Reply by Drarig29** [reply]
Is this something you really need? @KaranCK0 

IIRC the property on which the viewer relies to show this label is populated by the manager only _when it's not obvious_ (to quote my previous message), so both the manager and the viewer would have to be updated in order to have the information available (I prefer not to have too much complex logic in the viewer, and keep it as "dumb" as possible).

**Reply by KaranCK0** [reply]
We want feature winner bracket matches labels.

also we need to add date and time on the round name ( round 1 - 12 Jun, 2025) like this.
can we do this? @Drarig29

**Reply by Drarig29** [reply]
First, "want" is not polite. Second, please create a feature request in the Issues tab, and stop using this thread in an unrelated discussion. Thank you.
